_source/_posts/2020-12-09-dotnet-cloud-host-publish.md

```diff
@@ -0,0 +1,325 @@
+---
+layout: blog_post
+title: How to Deploy Your .NET Core App to Google Cloud, AWS or Azure
+author: greg-sinka
+by: contractor
+communities: [.net]
+description: "How to deploy a .NET Core app to AWS, Azure or Google Cloud."
+tags: [google-cloud, dotnet, aspnet, csharp, dotnetcore, aspnetcore, aws, azure, deployment]
+tweets:
+- "Learn how to deploy an Okta #dotnetcore app to #AWS, #GoogleCloud or #Azure."
+- "Discover how to deploy and configure a secured #dotnetcore app to #GoogleCloud, #Azure or #AWS!"
+- "Need to deploy your #dotnetcore to a major cloud host like #Azure #AWS or #GoogleCloud? Here you go!"
+image: blog/featured/okta-dotnet-mouse-down.jpg
+type: conversion
+---
+
+There has been a cut-throat competition between cloud hosts in the past few years - each attempting to earn the sympathy of developers and dev-ops by rolling out shiny new tools, plugins, and integrations. There are a gazillion how-to tutorials and guides in the community on using these tools. Sometimes when looking for a solution, it is hard to find the newest and simplest way. I never know if an article written last year is still valid using today's tools. In this article, I'm going to walk you through the deployment of a .NET Core 3.1 application to the major cloud hosts (Microsoft Azure App Service, AWS Beanstalk, Google Cloud App Engine) in the simplest possible way. After deployment, I'll take a few steps forward and configure Okta Authentication in the example application to show how to add a login flow at once.
+
+## What is Okta
+
+Okta is a cloud service that allows developers to create, edit, and securely store user accounts and user account data, and connect them with one or multiple applications. Using Okta you don't have to worry about implementing sign up, login, or logout flows manually.
+
+## Prerequisites
+
+[Okta Developer account](https://developer.okta.com/)
+[.NET Core 3.1 SDK](https://dotnet.microsoft.com/download)
+Any preferred code editor, I'll be using [VS Code](https://code.visualstudio.com/) for Mac 😱.
+
+*Azure deployment:*
+[Microsoft Azure](https://azure.microsoft.com) account
+[Azure CLI](https://docs.microsoft.com/en-us/cli/azure/install-azure-cli)
+*AWS deployment:*
+[AWS account](https://aws.amazon.com)
+
+*Google Cloud deployment:*
+[Google Cloud](https://cloud.google.com) account
+[Google Cloud SDK](https://cloud.google.com/sdk)
+
+## Preparing the .NET Core App for Cloud Host deployment
+
+I have prepared an example application to play around with and see how the deployments work.
+
+Start by cloning the repo to your local machine:
+
+```sh
+git clone https://github.com/oktadeveloper/okta-netcore3-deploy-to-cloud-hosts-example
```

Comment:
The clone command from GitHub has a .git on the end.

```
git clone https://github.com/oktadeveloper/okta-netcore3-deploy-to-cloud-hosts-example.git
```

However, this one seems to work, so not sure it needs to be changed.

Reply:
Related: the README for this repo needs instructions so folks can clone it, configure it, and run it w/o going anywhere else. I'm not sure if it's necessary to add it again at the bottom, but I tend to do that in my posts.

Reply:
I haven't gone over that update with any of the contractors yet